<launch>
  <node name="rviz" pkg="rviz" type="rviz"
    args="-d $(find crowdbot_active_slam)/rviz/crowdbot_active_slam.rviz"/>
  <!-- robot_name:
       pioneer_sim, pepper_real, turtlebot_real, scientifica-->
  <arg name="robot_name" default="scientifica"/>
  <arg name="scan_topic" default="robot_1/scan"/>
  <node name="graph_optimisation" pkg="crowdbot_active_slam"
    type="graph_optimisation" output="screen">
    <param name="robot_name" value="$(arg robot_name)"/>
    <param name="scan_callback_topic" value="$(arg scan_topic)"/>

    <param name="node_dist_linear" value="0.15"/>
    <param name="node_dist_angular" value="0.2"/>

    <!-- params for frontend scan matcher -->
    <param name="frontend_max_iterations" value="20"/>
    <param name="frontend_max_correspondence_dist" value="0.3"/>
    <param name="frontend_max_angular_correction_deg" value="45"/>
    <param name="frontend_max_linear_correction" value="0.5"/>
    <param name="frontend_outliers_maxPerc" value="0.90"/>
    <param name="frontend_orientation_neighbourhood" value="10"/>
    <param name="frontend_outliers_adaptive_order" value="0.7"/>
    <param name="frontend_do_compute_covariance" value="1"/>
    <!-- sigma = 0.02(sick lidar) + 0.01(combination/tf/reflection error) -->
    <param name="frontend_sigma" value="0.03"/>

    <!-- params for graph building and map -->
    <param name="loop_closing_radius" value="0.7"/>
    <param name="const_map_update_steps" value="false"/>
    <param name="map_width" value="10000"/>
    <param name="map_height" value="10000"/>
    <param name="map_resolution" value="0.01"/>

    <param name="time_thershold_frontend" value="0.10"/>
    <param name="time_thershold_loopclosing" value="0.15"/>
  </node>
</launch>
